packages/react-components/src/components/utils/ChatMessageComponentAsEditBoxUtils.tsx

```diff
@@ -57,3 +57,39 @@ export function getMessageState(
 export function getMessageWithAttachmentMetadata(message: ChatMessage): AttachmentMetadata[] | undefined {
   return message.attachments;
 }
+
+/* @conditional-compile-remove(attachment-upload) */
+/**
+ * @private
+ */
+interface Action {
+  type: string;
+}
+
+/* @conditional-compile-remove(attachment-upload) */
+/**
+ * @private
+ */
+type Actions = RemoveAction;
```

Comment:
This as plural Action's' is a bit misleading since its a single action when using as a parameter implies it may be an array or something

Reply:
good catch

Reply:
i see, this is actually for maintaining the state in edit box, currently we can only remove, but later on we will have clear. set, add and other actions. This is meant to be reserved so we can do a union later on:

```
type Actions = RemoveAction | SetAction | ClearAction ....;
```

added a comment to clarify the naming

Reply:
Plural should be here for extensibility

Reply:
Sorry I meant to be pointing out here rather than the type itself
export const attachmentMetadataReducer = (state: AttachmentMetadata[], action: Actions): AttachmentMetadata[] => {

`action: Actions` the naming is kind of odd a singular `action` but the type is plural Action"s", maybe name Action to ActionType or something?

Reply:
`Actions` is the type; it is an enum of multiple possible actions.
`action` is the singular action the reducer handles.